### PR TITLE
[PM-8225] Disable biometrics timeout check in development mode

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8225

## 📔 Objective

Improves the debugging experience for biometrics ipc by disabling the age check on messages when in dev mode. Since this is limited to development mode it does not impact security.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by quexten** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/bitwarden/clients/10052?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10052** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/10052?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#9979** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/9979?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9977** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/9977?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9976** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/9976?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9945** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/9945?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9944** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/9944?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#9943** <a href="https://app.graphite.dev/github/pr/bitwarden/clients/9943?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @quexten and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/browser/src/background/nativeMessaging.background.ts [hunk 1]
```diff
@@ -295,7 +295,11 @@ export class NativeMessagingBackground {
       );
     }
 
-    if (Math.abs(message.timestamp - Date.now()) > MessageValidTimeout) {
+    // Check to prevent replay of old messages. This is disabled in dev mode
+    if (
+      Math.abs(message.timestamp - Date.now()) > MessageValidTimeout &&
+      !this.platformUtilsService.isDev()
+    ) {
       this.logService.error("NativeMessage is to old, ignoring.");
```
Comment (dani-garcia) [inline apps/browser/src/background/nativeMessaging.background.ts:303]:
```suggestion
      this.logService.error("NativeMessage is too old, ignoring.");
```

File: apps/desktop/src/services/native-messaging.service.ts [hunk 2]
```diff
@@ -131,7 +131,11 @@ export class NativeMessagingService {
       return;
     }
 
-    if (Math.abs(message.timestamp - Date.now()) > MessageValidTimeout) {
+    // Check to prevent replay of old messages. This is disabled in dev mode
+    if (
+      Math.abs(message.timestamp - Date.now()) > MessageValidTimeout &&
+      !this.platformUtilService.isDev()
+    ) {
       this.logService.error("NativeMessage is to old, ignoring.");
```
Comment (dani-garcia) [inline apps/desktop/src/services/native-messaging.service.ts:139]:
```suggestion
      this.logService.error("NativeMessage is too old, ignoring.");
```

**Review by dani-garcia — APPROVED** [review]
LGTM!
Small nit, just noticed some typos in the log messages.

**Review by jprusik — APPROVED** [review]

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**